Dockerfile

```diff
@@ -53,13 +56,16 @@ RUN echo "@cloudposse https://apk.cloudposse.com/3.9/vendor" >> /etc/apk/reposit
 # Use TLS for alpine default repos
 RUN sed -i 's|http://dl-cdn.alpinelinux.org|https://alpine.global.ssl.fastly.net|g' /etc/apk/repositories && \
     echo "@testing https://alpine.global.ssl.fastly.net/alpine/edge/testing" >> /etc/apk/repositories && \
-    echo "@community https://alpine.global.ssl.fastly.net/alpine/edge/community" >> /etc/apk/repositories && \
-    apk update
+    echo "@community https://alpine.global.ssl.fastly.net/alpine/edge/community" >> /etc/apk/repositories
 
 # Install alpine package manifest
 COPY packages.txt /etc/apk/
 
-RUN apk add --update $(grep -v '^#' /etc/apk/packages.txt) && \
+
+# Import package version for cache busting
+ARG CLOUDPOSSE_PACKAGES_VERSION
+RUN echo Using cloudposse/packages version ${CLOUDPOSSE_PACKAGES_VERSION} as packages cache key && \
```

Comment:
Aha, I see... So that forces this to re-run. What if we used `ADD ... https://apk....APKINDEX` instead? I think `ADD` will invalidate the cache if the contents of the source change.

Reply:
What I don't like about this hack is that there's actually nothing more than optimistic correlation between `CLOUDPOSSE_PACKAGES_VERSION` and the packages installed.